src/gui/qgsattributeform.cpp

```diff
@@ -658,6 +662,9 @@ void QgsAttributeForm::onAttributeChanged( const QVariant &value )
   if ( oldValue == value && oldValue.isNull() == value.isNull() )
```

Comment:
Btw. I wonder about the correctness of this check, because it doesn't compare against the previously entered value, but against the original value, so that the code below isn't executed when the user changed the value and then changes it back to the original value.